File: src/LinksNews/angular2App/app/core/domain/page/page.ts
import { Column } from "./column";
import { Account } from "../account";
import * as _ from 'lodash';
import { ContentCategory } from "../content-category";

export class Page {
    id: number;
    accountId: number;
    name: string;
    title: string;
    publicAccess: boolean = false;
    imageUrl: string;
    description: string;
    pageIndex: number;
    showTitle: boolean = true;
    showDescription: boolean = true;
    showImage: boolean = true;
    showAd: boolean = false;
    adContent: string;
    columns: Array<Column> = new Array<Column>();
    login: string;
    selected: boolean = false;
    categories: Array<ContentCategory> = new Array<ContentCategory>();

    get checkedCategoriesCount(): number {
        if (!this.categories) {
            return 0;
        }
        let i: number = 0;
        this.categories.forEach(x => {
            if (x.selected) {
                i++;
            }
        });    
        return i;
    }

    clone(): Page {

        let result: Page = <Page>_.assign(new Page(), this);

        if (this.columns) {
            result.columns = new Array<Column>();
            this.columns.forEach(x => result.columns.push(x.clone()));
        }

        if (this.categories) {
            result.categories = new Array<ContentCategory>();
            this.categories.forEach(x => result.categories.push(x.clone()));
        }

    
        return result;
    }

    cloneNoColumnsChildren(): Page {

        let result: Page = <Page>_.assign(new Page(), this);

        if (this.columns) {
            result.columns = new Array<Column>();
            this.columns.forEach(x => result.columns.push(x.cloneNoChildren()));
        }

        if (this.categories) {
            result.categories = new Array<ContentCategory>();
            this.categories.forEach(x => result.categories.push(x.clone()));
        }
    
        return result;
    }

    static from(page: Page): Page {

        let result: Page = new Page();

        _.assign(result, page);

        result.columns = new Array<Column>();
        
        if (page.columns) {
            page.columns.forEach(x => result.columns.push(Column.from(x)));
        }

        result.categories = new Array<ContentCategory>();

        if (page.categories) {
            page.categories.forEach(x => result.categories.push(ContentCategory.from(x)));
        }

        return result;
    }

}

File: src/LinksNews/Services/Data/Models/Page.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace LinksNews.Services.Data.Models
{
    [Table("page")]
    public class Page
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public long? Id { get; set; }
        public long? AccountId { get; set; }
        public string Name { get; set; }
        public string Title { get; set; }
        public bool PublicAccess { get; set; }
        public string Description { get; set; }
        public long? PageIndex { get; set; }
        public bool ShowTitle { get; set; }
        public bool ShowDescription { get; set; }
        public bool ShowImage { get; set; }

        [NotMapped]
        public string ImageUrl { get; set; }

        [NotMapped]
        public bool ShowAd { get; set; }

        [NotMapped]
        public string AdContent { get; set; }

        [NotMapped]
        public string Login { get; set; }

        [NotMapped]
        public virtual List<Column> Columns { get; set; }

        [NotMapped]
        public virtual List<ContentCategory> Categories { get; set; }
    }
}
